pkg/okteto/client.go

```diff
@@ -96,11 +102,26 @@ func NewOktetoClientFromUrl(url string) (*OktetoClient, error) {
 		return nil, err
 	}
 
-	httpClient := oauth2.NewClient(context.Background(), nil)
+	ctx := backgroundContextWithHttpClient(config.IsInsecureSkipVerify())
+	httpClient := oauth2.NewClient(ctx, nil)
 
 	return newOktetoClientFromGraphqlClient(u, httpClient)
 }
 
+func backgroundContextWithHttpClient(insecureSkipVerify bool) context.Context {
+	return context.WithValue(
+		context.Background(),
+		oauth2.HTTPClient,
+		&http.Client{
+			Transport: &http.Transport{
+				TLSClientConfig: &tls.Config{
+					InsecureSkipVerify: insecureSkipVerify,
```

Comment:
```suggestion
					InsecureSkipVerify: insecureSkipVerify, // skipcq: GSC-G402
```

can we add this so deepsource does not take this as a flag?